### PR TITLE
Python asyncio reactor rebase

### PR DESCRIPTION
**Pull request by mambocab**
Look good, folks? @bjmb @aboudreault 

Here's a successful test run with the new reactor: http://jenkins-devtools.datastax.lan/view/python-driver/job/datastax.python-driver.python-asyncio-reactor-rebase.adhoc_asyncio_integration/1/

If we want, we can rebase `gate import on run_coroutine_threadsafe` and `document AsyncioConnection as experimental` into the main `asyncio` commit before merge.

### CODE REVIEW

File: tests/__init__.py [hunk 1]
```diff
@@ -86,7 +89,7 @@ def is_monkey_patched():
         connection_class = None
 
 
-MONKEY_PATCH_LOOP = bool(os.getenv('MONKEY_PATCH_LOOP', False))
+MONKEY_PATCH_LOOP = os.getenv('MONKEY_PATCH_LOOP', False)
```
[the remark this answers is not in the document]
Comment (mambocab) [reply tests/__init__.py]: Filed https://datastax-oss.atlassian.net/browse/PYTHON-903

File: tests/unit/io/test_asyncioreactor.py [hunk 2]
```diff
@@ -0,0 +1,65 @@
+
+try:
```
[the remark this answers is not in the document]
Comment (mambocab) [reply tests/unit/io/test_asyncioreactor.py]: Thanks! Yeah, that `−715` feels good for sure

**Comment by mambocab** [comment]
Addressed [this comment](https://github.com/datastax/python-driver/pull/907#discussion_r163048530) and rebased from https://github.com/datastax/python-driver/pull/907#discussion_r163048530

**Comment by hellysmile** [comment]
Hey guys! It is extremely good new that asyncio reactor is coming cassandra-driver, but as I see it do not allow yet to execute asynchronous queries like

```python
yield from session.execute('SELECT now() FROM system.local;')
```

I am working for a while on https://github.com/aio-libs/aiocassandra

It is very simple thread future / asyncio futures abstraction on `session.execute_async`

Does it makes sense to point in documentation aiocassandra?

Some new features will be released soon
https://github.com/aio-libs/aiocassandra/pull/14 
`async for` support over pagination of responses.

**Comment by mambocab** [comment]
@hellysmile Hey, thanks for the pointer -- `aiocassandra` is really cool, and we love to see community projects that extend the driver. However, we don't link to non-DataStax tooling in our READMEs, since we don't want to imply that DataStax produces or provides support for it.

That said, thanks for putting time into making community tooling around the driver -- `async for` support looks awesome!

**Comment by hellysmile** [comment]
What about to merge aiocassandra into python-driver?

**Comment by mambocab** [comment]
@hellysmile We're still figuring out what we want `asyncio` support to look like going forward. If it ends up looking like a wrapper on top of the existing interface, we'll be in touch, since `aiocassandra` looks like a great implementation of that.

Sorry to give such an indefinite answer, but we're fairly new to `asyncio` as a team and are still exploring it.

**Comment by sandrotosi** [comment]
i cant report a bug report via jira because 

```
Your email address email@domain.ext doesn't have access because datastax-oss.atlassian.net has reached its license limit.

You can request access from your site admin.```

but i just want to report here that the introduction of `yield from` breaks the code for python2.7. `cassandra/io/asyncioreactor.py` is installed both if you run `python setup.py install` and `python3 setup.py install`  but it's only compatible with python3, which generates:

```
  File "/usr/lib/python2.7/dist-packages/cassandra/io/asyncioreactor.py", line 51
    yield from asyncio.sleep(timeout, loop=loop)
             ^
SyntaxError: invalid syntax
```

this is preventing debian to update this package in the distribution

**Comment by aboudreault** [comment]
@sandrotosi We are looking at the Jira issue.

Thanks for the report, I do see the error during the byte compiling. However, the package seems to be usable. I will create a ticket to fix this!

**Comment by sandrotosi** [comment]
thanks! in debian we byte-compile all the python code at install time (to avoid inconsistent import timing) so this fails when we install the package on our systems

**Comment by aboudreault** [comment]
Good to know Sandro, we'll keep that in mind.

**Comment by aboudreault** [comment]
For the record: https://datastax-oss.atlassian.net/browse/PYTHON-994

**Pull request merged**